# Exploration and Analysis of Mental Health Data from Shamiri

### by F Njakai

### Table of contents
* [Introduction](#introduction)
* [Preliminary Wrangling](#preliminary-wrangling)
* [Exploration](#exploration)
* [Summary of Findings](#exploration)
* [Conclusions](#conclusions)

<div id="introduction"></div>

## Introduction

<div id="preliminary-wrangling"></div>

## Preliminary Wrangling

In [4]:
#install and/or upgrade packages

!pip install pandas seaborn pandas-profiling

Defaulting to user installation because normal site-packages is not writeable


In [20]:
#imports

import warnings
warnings.simplefilter(action='ignore', category=FutureWarning)
warnings.filterwarnings(action='ignore')
import numpy as np
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt
import pandas_profiling
from random import randint
from os import path
from contextlib import suppress
from pandas_profiling import ProfileReport

%load_ext autoreload
%autoreload 2
%reload_ext autoreload


%matplotlib inline

The autoreload extension is already loaded. To reload it, use:
  %reload_ext autoreload


### Default settings for plots

automate, as much as possible, the process of creating visualisations

##### why?
* it is efficient
* visualisations are consistent

##### how?
* create templates

In [2]:
#template no. 1

#default blue
default_blue = sns.color_palette('tab10')[0]

#default orange
default_orange = sns.color_palette('tab10')[1]

#default palette
default_palette = sns.color_palette('tab10')

In [3]:
#template no. 2

'''
Simple function to create `Figure` object
using matplotlib. Has an x-lab, y-lab and
title.

"Father Figure", if you like :)

3 params, all type `str`:
@x_lab: x label
@y_lab: y label
@title: title

return: None
'''

def create_fig(x_lab: str, y_lab: str, title: str):
    """Father Figure"""
    try:
        plt.figure(figsize=(10, 6.18), dpi=216, frameon=False, clear=True)
        plt.xlabel(x_lab)
        plt.ylabel(y_lab)
        plt.title(title)
    except ModuleNotFoundError:
        print(f'Please `import matplotlib.pyplot as plt` and try again')
    except:
        print(f'Failed to create template')
        raise

In [4]:
#template no. 3

'''
Simple function to create `Figure` object
using matplotlib for sub-plots.

"Father Figure", for sub-plots :)

2 params, type `int`; number of sub-plots:
@n_row: #rows
@n_col: #cols

return: fig and ax objects
'''

def create_sub(n_row: int=1, n_col: int=1):
    """Father Figure for  sub-plots"""
    try:
        fig, ax = plt.subplots(n_row, n_col, figsize=(10, 6.18), dpi=216)
        fig.tight_layout(pad=10.0)
        return fig, ax
    except ModuleNotFoundError:
        print(f'Please `import matplotlib.pyplot as plt` and try again')
    except:
        print(f'Failed to create template')
        raise

In [5]:
#confirm that a df exists

'''
Simple function to see if a df exists

1 param: name of variable holding the df

Do not pass the arg as a string. 
Repeat: DO NOT pass the arg as a string.

return: None
'''

def confirm_df_exists(df):
    """confirm that a df exists"""
    if not df.empty:
        print(f'This dataframe exists')
        return
    print(f'This dataframe does not exist')

In [6]:
#confirm that a file exists and/or has been
# created in current dir

'''
simple function to confirm that a file exists
and/or has been created in current dir

1 param, type `str`: name of file
@file_name: name of file

return None
'''

def confirm_file_exists(file_name: str):
    """confirm that file exists"""
    if path.exists(file_name):
        print(f'File exists')
    else:
        print(f'Something went wrong. Investigate')

### Load the data

In [16]:
df = pd.read_csv('shamiri_imputed_dataset.csv', sep=',')
confirm_df_exists(df)

This dataframe exists


In [23]:
profile = ProfileReport(df)
profile.to_file(output_file='report.html')

Export report to file: 100%|██████████| 1/1 [00:00<00:00, 34.52it/s]


In [28]:
df.shape

(658, 33)

In [25]:
df.duplicated().value_counts()

False    658
dtype: int64

In [26]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 658 entries, 0 to 657
Data columns (total 33 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   ParticipantID     658 non-null    object 
 1   PHQ1              658 non-null    int64  
 2   PHQ2              658 non-null    int64  
 3   PHQ3              658 non-null    int64  
 4   PHQ4              658 non-null    int64  
 5   PHQ5              658 non-null    int64  
 6   PHQ6              658 non-null    int64  
 7   PHQ7              658 non-null    int64  
 8   PHQ8              658 non-null    int64  
 9   GAD1              658 non-null    int64  
 10  GAD2              658 non-null    int64  
 11  GAD3              658 non-null    int64  
 12  GAD4              658 non-null    int64  
 13  GAD5              658 non-null    int64  
 14  GAD6              658 non-null    int64  
 15  GAD7              658 non-null    int64  
 16  MSSS1             658 non-null    int64  
 1

In [31]:
df.ParticipantID.nunique()

626

In [35]:
df.duplicated().apply(lambda x: x != 'True').sample(randint(5, 15))

476    True
578    True
35     True
10     True
599    True
382    True
205    True
54     True
401    True
301    True
408    True
324    True
597    True
194    True
dtype: bool

In [27]:
df.sample(randint(5, 15))

ParticipantID  PHQ1  PHQ2  PHQ3  PHQ4  PHQ5  PHQ6  PHQ7  PHQ8  GAD1  ...  \
137        EH_058     0     1     0     0     0     2     0     2     0  ...   
572       OLY_127     1     0     0     1     2     0     2     0     2  ...   
470       OLY_025     1     1     2     1     0     1     0     0     0  ...   
430      AGHS_142     2     1     1     2     0     1     1     3     1  ...   
520       OLY_075     1     0     0     3     0     0     2     0     0  ...   
620       OLY_175     2     3     1     0     1     1     1     1     0  ...   
271       AHS_115     3     0     0     3     1     1     3     3     2  ...   
120        EH_041     2     2     2     2     2     3     2     2     2  ...   

     MSSS8  MSSS9  MSSS10  MSSS11  MSSS12     Tribe  Gender   School   Age  \
137      7      4       6       7       4  Minority       F    Elite  17.0   
572      2      3       2       3       3  Minority       F  Olympic  14.0   
470      2      1       5       6       5  Minority       M  Olympic  14.0   
430      4      6       6       4       5  Majority       F     AGHS  15.0   
520      7      7       7       7       7  Minority       F  Olympic  16.0   
620      6      6       6       6       6  Minority       M  Olympic  17.0   
271      1      5       3       1       4  Majority       M      AHS  17.0   
120      6      3       7       4       2  Minority       M    Elite  16.0   

     School_Resources  
137              Poor  
572            Medium  
470            Medium  
430              Rich  
520            Medium  
620            Medium  
271              Rich  
120              Poor  

[8 rows x 33 columns]

### Structure

#### Overall
* 657 observations
* 32 variables
    * 27 of type `int`
    * 1 of type `float`
    * 4 of type `object`

#### Missing and null values
* `df` has no missing or null values

#### Duplicated observations
* `df` has no duplicated observations

#### Multiple values for a variable
* observations in twtr_archive have a single value per variable

#### Quality
* no missing values
* values follow the [definition of variables](./variables.md)

#### Tidiness
* variables are properly defined
* each variable represents one piece of information
* observations in twtr_archive have a single value per variable



<div id="explration"></div>

## Exploration

<div id="summary-of-findings"></div>

## Summary of Findings

<div id="conclusions"></div>

## Conclusions